### PR TITLE
Fix slo-monitor dependencies

### LINKED ISSUE #1481: Failing pull-perf-tests-verify-{all,lint} presubmit jobs
**Which jobs are failing**:
pull-perf-tests-verify-lint
pull-perf-tests-verify-all

**Since when has it been failing**:
Sep 18: https://prow.k8s.io/job-history/gs/kubernetes-jenkins/pr-logs/directory/pull-perf-tests-verify-all

**Reason for failure**:
`slo-monitor` vendors old Prometheus package which leverages `goautoneg` package hosted at bitbucket.org using Mercurial which was recently removed (https://bitbucket.org/blog/sunsetting-mercurial-support-in-bitbucket).

**Anything else we need to know**:
I'm currently working on fixing that.

/assign

### PR DESCRIPTION
**Pull request by tosi3k**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #1481

**Special notes for your reviewer**:
/cc @mborsz 

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: slo-monitor/src/main/slo-monitor.go [hunk 1]
```diff
@@ -27,7 +27,7 @@ import (
 	"k8s.io/client-go/tools/clientcmd"
 
 	"github.com/golang/glog"
-	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
```
Comment (tosi3k) [inline slo-monitor/src/main/slo-monitor.go:30]: FTR: https://github.com/prometheus/client_golang/pull/214

**Comment by mm4tt** [comment]
Rubber stump approval to not block this. Please get it reviewed by @mborsz 

/approve
/hold

File: slo-monitor/go.mod [hunk 2]
```diff
@@ -1,15 +1,9 @@
...
 require (
-	bitbucket.org/ww/goautoneg v0.0.0-20120707110453-75cd24fc2f2c // indirect
-	github.com/beorn7/perks v0.0.0-20160229213445-3ac7bf7a47d1 // indirect
 	github.com/golang/glog v0.0.0-20141105023935-44145f04b68c
 	github.com/googleapis/gnostic v0.2.0 // indirect
-	github.com/matttproud/golang_protobuf_extensions v0.0.0-20150406173934-fc2b8d3a73c4 // indirect
-	github.com/prometheus/client_golang v0.0.0-20151008143247-e51041b3fa41
-	github.com/prometheus/client_model v0.0.0-20150212101744-fa8ad6fec335 // indirect
-	github.com/prometheus/common v0.0.0-20151014055050-ffe929a3f4c4 // indirect
-	github.com/prometheus/procfs v0.0.0-20150928173926-454a56f35412 // indirect
```
Comment (tosi3k) [inline slo-monitor/go.mod:12]: Shall we remove those guys from the `vendor` subdirectory on the occasion?

Comment (wojtek-t) [reply slo-monitor/go.mod:12]: Yes - you need to regenerate vendor/ (or get rid of it completely).

Comment (mborsz) [reply slo-monitor/go.mod:12]: Yes, I think you need to run go mod tidy and go mod vendor here.

We should remove vendor as well, but maybe in a separate PR?

Comment (tosi3k) [reply slo-monitor/go.mod:12]: I'll get rid of it in a follow-up PR.

Comment (wojtek-t) [reply slo-monitor/go.mod:12]: Sure - it shouldn't be the same PR. It should be either before this PR or after.

Comment (tosi3k) [reply slo-monitor/go.mod:12]: Done.

**Comment by mm4tt** [comment]
/lgtm

Let's merge this given that it fixes the broken tests. The broken tests block other PRs.

**Pull request merged**